packages/core/src/external/athenahealth/index.ts

```diff
@@ -0,0 +1,331 @@
+import axios, { AxiosInstance, AxiosResponse, AxiosError } from "axios";
+import {
+  Medication,
+  MedicationAdministration,
+  MedicationDispense,
+  MedicationStatement,
+} from "@medplum/fhirtypes";
+import {
+  patientResourceSchema,
+  PatientResource,
+} from "@metriport/shared/interface/external/athenahealth/patient";
+import { errorToString } from "@metriport/shared";
+import { S3Utils } from "../aws/s3";
+import { out } from "../../util/log";
+import { capture } from "../../util/notifications";
+import { Config } from "../../util/config";
+import { uuidv7 } from "../../util/uuid-v7";
+import { createHivePartitionFilePath } from "../../domain/filename";
+
+interface ApiConfig {
+  threeLeggedAuthToken: string;
+  practiceId: string;
+  environment: "api" | "api.preview";
+  clientKey: string;
+  clientSecret: string;
+}
+
+const region = Config.getAWSRegion();
+const responsesBucket = Config.getEhrResponsesBucketName();
+
+function getS3UtilsInstance(): S3Utils {
+  return new S3Utils(region);
+}
+
+export type AthenaMedication = { medication: string; medicationid: number };
+
+export type MedicationWithRefs = {
+  medication: Medication;
+  administration?: MedicationAdministration;
+  dispense?: MedicationDispense;
+  statement?: MedicationStatement;
+};
+
+class AthenaHealthApi {
+  private axiosInstanceFhirApi: AxiosInstance;
+  private axiosInstanceProprietary: AxiosInstance;
+  private twoLeggedAuthToken: string;
+  private threeLeggedAuthToken: string;
+  private practiceId: string;
+  private s3Utils: S3Utils;
+
+  private constructor(private config: ApiConfig) {
+    this.twoLeggedAuthToken = "";
+    this.threeLeggedAuthToken = config.threeLeggedAuthToken;
+    this.practiceId = config.practiceId.replace("a-1.Practice-", "");
+    this.s3Utils = getS3UtilsInstance();
+    this.axiosInstanceFhirApi = axios.create({});
+    this.axiosInstanceProprietary = axios.create({});
+  }
+
+  public static async create(config: ApiConfig): Promise<AthenaHealthApi> {
+    const instance = new AthenaHealthApi(config);
+    await instance.initialize();
+    return instance;
+  }
+
+  private async fetchtwoLeggedAuthToken(): Promise<void> {
+    const url = `https://${this.config.environment}.platform.athenahealth.com/oauth2/v1/token`;
```

Comment:
everywhere: prefer to put `https://${this.config.environment}.platform.athenahealth.com/` in a const called `BASE_URL` or something so we don't have to change in 3 places if needed

Reply:
Done.